python/cudf/cudf/core/column/string.py

```diff
@@ -399,6 +406,175 @@ def lower(self):
             self._parent.nvstrings.lower(), index=self._index, name=self._name
         )
 
+    def slice(self, start=None, stop=None, step=None):
+        """
+        Returns a substring of each string.
+
+        Parameters
+        ----------
+        start : int
+            Beginning position of the string to extract.
+            Default is beginning of the each string.
+        stop : int
+            Ending position of the string to extract.
+            Default is end of each string.
+        step : int
+            Characters that are to be captured within the specified section.
+            Default is every character.
+
+        Returns
+        -------
+        Series/Index of str dtype
+            A substring of each string.
+
+        """
+
+        from cudf.core import Series
+
+        if start is None:
+            start = 0
+        if stop is None:
+            stop = -1
+        if step is None:
+            step = 1
+
+        return Series(
```

Comment:
We should return a column here instead so this can be reused for Series / Index as needed.